Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,866 @@
+              
+              NSArray *subnodes = [child nodesForXPath: @"//application" error: NULL];
+              NSXMLNode *appNode = [subnodes objectAtIndex: 0];
+              [self processChild: child
+                         withDoc: doc
+                     withAppNode: appNode
+                         sceneId: sceneId];
+              
+              // fix other custom objects
+              document = [[NSXMLDocument alloc] initWithRootElement: doc]; // put it into the document, so we can use Xpath.
+              controllerId = [self controllerIdWithDocument: document];
+              RELEASE(doc);
+
+              NSArray *customObjects = [document nodesForXPath: @"//objects/customObject" error: NULL];
+              FOR_IN(NSXMLElement*, coel, customObjects)
+                {
+                  [self processCustomElement: coel
+                                 forDocument: document];
+                }
+              END_FOR_IN(customObjects);
+
+              // Create document...
+              [_scenesMap setObject: document
+                             forKey: sceneId];
+              
+              // Map controllerId's to scenes...
+              if (controllerId != nil)
+                {
+                  [_controllerMap setObject: sceneId
+                                     forKey: controllerId];
+                }
+              
+              RELEASE(document);
+            }
+          END_FOR_IN(children);
+        }
+      END_FOR_IN(array);
+    }
+  else
+    {
+      NSLog(@"No document element in storyboard file");
+    }
+}
+
+- (NSString *) findControllerIdInDocument: (NSXMLDocument *)xmlIn
+                              withObjects: (NSXMLElement *)objects
+{
+  NSString *src = nil;
+  NSArray *controllers = [objects nodesForXPath: @"windowController"
+                                          error: NULL];
+  if ([controllers count] > 0)
+    {
+      NSXMLElement *controller = (NSXMLElement *)[controllers objectAtIndex: 0];
+      NSXMLNode *idAttr = [controller attributeForName: @"id"];
+      src = [idAttr stringValue];
+    }
+  else
+    {
+      controllers = [objects nodesForXPath: @"viewController"
+                                     error: NULL];
+      if ([controllers count] > 0)
+        {
+          NSXMLElement *controller = (NSXMLElement *)[controllers objectAtIndex: 0];
+          NSXMLNode *idAttr = [controller attributeForName: @"id"];
+          src = [idAttr stringValue];
+        }
+    }
+  return src;
+}
+
+- (BOOL) isProcessedDocument: (NSXMLDocument *)xmlIn
+                        uuid: (NSMutableString *)uuid
+{
+  NSArray *docArray = [xmlIn nodesForXPath: @"document" error: NULL];
+  if ([docArray count] > 0)
+    {
+      NSXMLElement *docElem = (NSXMLElement *)[docArray objectAtIndex: 0];
+      NSXMLNode *a = [docElem attributeForName: @"uuid"];
+      NSString *value = [a stringValue];
+      if (value != nil)
+        {
+          [uuid appendString: value];
```

Comment:
You never use the uuid string in the case where this method return YES. Maybe you could turn the uuid into the return value and only return it when a new one gets created? Or could all of that be simplified even more?

Reply:
Currently if this is set, then it is assumed that the document has been processed.  I suppose I could replace this with just a "true" or "false" value.

Reply:
Done.